apps/frontend/src/libs/helpers/rings.helper.ts

```diff
@@ -0,0 +1,174 @@
+import { LandingBgRingType, LandingBgSide } from "~/libs/enums/enums.js";
+import {
+	LandingBgInit,
+	LandingBgNumeric,
+	LandingBgPhysics,
+	LandingBgRingMinSpeed,
+	LandingBgRingSize,
+	LandingBgRingSpeedScale,
+	type RingConfig,
+	type RingType,
+} from "~/modules/landing/landing.js";
+
+import { clamp } from "./helpers.js";
+
+const safeRandom = (): number => {
+	const array = new Uint32Array(LandingBgNumeric.UINT32_ARRAY_LENGTH);
+	crypto.getRandomValues(array);
+
+	const value = array[LandingBgNumeric.ZERO] ?? LandingBgNumeric.ZERO;
+
+	return value / (LandingBgNumeric.UINT32_MAX + LandingBgNumeric.ONE);
+};
+
+const ensureMinSpeed = (value: number, minValue: number): number => {
+	if (Math.abs(value) < minValue) {
+		return value < LandingBgNumeric.ZERO ? -minValue : minValue;
+	}
+
+	return value;
```

Comment:
Consider early return
```suggestion
	if (Math.abs(value) >= minValue) {
		return value;
	}

	return value < LandingBgNumeric.ZERO ? -minValue : minValue;
```